element_calcium_imaging/export/nwb/nwb.py

```diff
@@ -0,0 +1,339 @@
+from datajoint import DataJointError
+from pynwb import NWBHDF5IO, NWBFile
+from pynwb.ophys import (
+    Fluorescence,
+    ImageSegmentation,
+    OpticalChannel,
+    RoiResponseSeries,
+    TwoPhotonSeries,
+)
+
+from ... import scan, imaging_no_curation
+from ...scan import get_calcium_imaging_files, get_imaging_root_data_dir
+
+
+logger = dj.logger
+
+if imaging_no_curation.schema.is_activated():
+    imaging = imaging_no_curation
+else:
+    raise DataJointError(
+        "This export function is designed for the `imaging_no_curation` module."
+    )
+
+
+def imaging_session_to_nwb(
+    session_key,
+    include_raw_data=False,
+    lab_key=None,
+    project_key=None,
+    protocol_key=None,
+    nwbfile_kwargs=None,
+):
+    """Main function for converting calcium imaging data to NWB.
+
+    Args:
+        session_key (dict): key from Session table.
+        include_raw_data (bool): Optional. Default False. Include the raw data from
+        source. `ScanImage`, `Scanbox`, and `PrairieView` are supported.
+        lab_key (dict): Optional key to add metadata from Element Lab.
+        project_key (dict): Optional key to add metadata from Element Lab.
+        protocol_key (dict): Optional key to add metadata from Element Lab.
+        nwbfile_kwargs (dict): Optional. If Element Session is not used, this argument
+            is required and must be a dictionary containing 'session_description' (str),
+            'identifier' (str), and 'session_start_time' (datetime), the required
+             minimal data for instantiating an NWBFile object. If element-session is
+             being used, this argument can optionally be used to overwrite NWBFile
+             fields.
+    Returns:
+        nwbfile (NWBFile): nwb file
+    """
+
+    session_to_nwb = getattr(imaging._linking_module, "session_to_nwb", False)
+
+    if session_to_nwb:
+        nwb_file = session_to_nwb(
+            session_key,
+            lab_key=lab_key,
+            project_key=project_key,
+            protocol_key=protocol_key,
+            additional_nwbfile_kwargs=nwbfile_kwargs,
+        )
+    else:
+        nwb_file = NWBFile(**nwbfile_kwargs)
+
+    if include_raw_data:
+        _create_raw_data_nwbfile(session_key, linked_nwb_file=nwb_file)
+
+        if not nwb_file.imaging_planes:
+            imaging_plane = _add_scan_to_nwb(session_key, nwbfile=nwb_file)
+
+        _add_image_series_to_nwb(
+            session_key, imaging_plane=nwb_file.imaging_planes["ImagingPlane"]
+        )
+        _add_segmentation_data_to_nwb(
+            session_key,
+            nwbfile=nwb_file,
+            imaging_plane=nwb_file.imaging_planes["ImagingPlane"],
+        )
+
+    else:
+        imaging_plane = _add_scan_to_nwb(session_key, nwbfile=nwb_file)
+        _add_image_series_to_nwb(session_key, imaging_plane=imaging_plane)
+        _add_segmentation_data_to_nwb(
+            session_key, nwbfile=nwb_file, imaging_plane=imaging_plane
+        )
```

Comment:
It looks like you could refactor the `_add_scan_to_nwb()`, `_add_image_series_to_nwb()`, `_add_segmentation_data_to_nwb()` to be outside of the `if/else` block

Reply:
Good catch! I just made this update in the most recent commit and tested it with and without raw data for a `ScanImage` tif.